src/rpc/misc.cpp

```diff
@@ -32,176 +32,71 @@
-        return obj;
-    }
-};
-#endif
-
 UniValue validateaddress(const JSONRPCRequest& request)
 {
     if (request.fHelp || request.params.size() != 1)
         throw std::runtime_error(
             "validateaddress \"address\"\n"
             "\nReturn information about the given bitcoin address.\n"
+            "DEPRECATION WARNING: Parts of this command have been deprecated and moved to getaddressinfo. Clients must\n"
+            "transition to using getaddressinfo to access this information before upgrading to v0.17. To have those deprecated\n"
+            "fields shown, start bitcoind with -deprecatedrpc=validateaddress"
             "\nArguments:\n"
-            "1. \"address\"     (string, required) The bitcoin address to validate\n"
+            "1. \"address\"                    (string, required) The bitcoin address to validate\n"
             "\nResult:\n"
             "{\n"
             "  \"isvalid\" : true|false,       (boolean) If the address is valid or not. If not, this is the only property returned.\n"
-            "  \"address\" : \"address\", (string) The bitcoin address validated\n"
+            "  \"address\" : \"address\",        (string) The bitcoin address validated\n"
             "  \"scriptPubKey\" : \"hex\",       (string) The hex encoded scriptPubKey generated by the address\n"
-            "  \"ismine\" : true|false,        (boolean) If the address is yours or not\n"
-            "  \"iswatchonly\" : true|false,   (boolean) If the address is watchonly\n"
             "  \"isscript\" : true|false,      (boolean) If the key is a script\n"
-            "  \"script\" : \"type\"             (string, optional) The output script type. Possible types: nonstandard, pubkey, pubkeyhash, scripthash, multisig, nulldata, witness_v0_keyhash, witness_v0_scripthash\n"
-            "  \"hex\" : \"hex\",                (string, optional) The redeemscript for the p2sh address\n"
-            "  \"addresses\"                   (string, optional) Array of addresses associated with the known redeemscript\n"
+            "  \"ismine\" : true|false,        (boolean) DEPRECATED. If the address is yours or not\n"
+            "  \"iswatchonly\" : true|false,   (boolean) DEPRECATED. If the address is watchonly\n"
+            "  \"iswitness\" : true|false,     (boolean) DEPRECATED. If the address is a witness address\n"
+            "  \"witness_version\" : version   (numeric, optional) DEPRECATED. The version number of the witness program\n"
+            "  \"witness_program\" : \"hex\"   (string, optional) DEPRECATED. The hex value of the witness program\n"
+            "  \"isscript\" : true|false,      (boolean) DEPRECATED. If the key is a script\n"
+            "  \"redeemscript\" : \"type\"       (string, optional) DEPRECATED. The type of the redeemscript. Possible types: nonstandard, pubkey, pubkeyhash, scripthash, multisig, nulldata, witness_v0_keyhash, witness_v0_scripthash\n"
```

Comment:
If the deprecation switch is enabled, should `redeemscript` be named `script` for compatibility in the output of `validateaddress`? Otherwise its a breaking change anyway

Reply:
Done